src/DiffEqUncertainty.jl

```diff
@@ -1,18 +1,38 @@
 module DiffEqUncertainty
 
-using DiffEqBase, Statistics, Distributions, Reexport
+# LinearAlgebra
+using DiffEqBase, Statistics, Reexport, RecursiveArrayTools, StaticArrays,
+    Distributions, KernelDensity, Zygote, LinearAlgebra, Random
+using Parameters: @unpack
+
 @reexport using Quadrature
-using KernelDensity
+import DiffEqBase: solve
+
+include("expectation/system_utils.jl")
+include("expectation/distribution_utils.jl")
+include("expectation/problem_types.jl")
+include("expectation/expectation.jl")
 
 include("probints.jl")
-include("koopman.jl")
+
 
 # Type Piracy, should upstream
 Base.eltype(K::UnivariateKDE)  = eltype(K.density)
 Base.minimum(K::UnivariateKDE) = minimum(K.x)
 Base.maximum(K::UnivariateKDE) = maximum(K.x)
+Base.extrema(K::UnivariateKDE) = minimum(K), maximum(K)
+
+Base.minimum(d::AbstractMvNormal) = fill(-Inf, length(d))
```

Comment:
These lines are technically type piracy right? Also, they might not be correct for any `AbstractNormal` that accepts singular covariance matrices, e.g., `Diagonal([0, 1])` does not have infinite support for the first variable.

Reply:
Yes, the `UnivariateKDE` parts should be upstreamed. `Base.minimum(d::AbstractMvNormal)` was upstream in Distributions in https://github.com/JuliaStats/Distributions.jl/pull/1319, probably need to double check that convo to makes sure a bug was introduced.